File: Java/javaperformancetester/src/main/java/edu/imepac/javaperformancetester/services/EncosteService.java
package edu.imepac.javaperformancetester.services;

import edu.imepac.javaperformancetester.models.Encoste;
import edu.imepac.javaperformancetester.models.Vagao;
import edu.imepac.javaperformancetester.repositories.EncosteRepository;
import edu.imepac.javaperformancetester.repositories.VagaoRepository;
import jakarta.transaction.Transactional;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;


@Service
public class EncosteService {

    @Autowired
    private EncosteRepository encosteRepository;

    @Autowired
    private VagaoRepository vagaoRepository;

    @Transactional
    public List<Encoste> salvarTodos(List<Encoste> encostes){

        List<Encoste> salvos= encosteRepository.saveAll(encostes);

        for(Encoste encoste : salvos){
            for(Vagao vagao : encoste.getVagoes()){
                vagao = vagaoRepository.findById(vagao.getId()).orElse(null);
            }
        }
        return encosteRepository.saveAll(encostes);
    }

    public List<Encoste> findAll(){
        return encosteRepository.findAll();
    }

    public Encoste create(Encoste encoste){
        return encosteRepository.save(encoste);
    }
}

File: Kotlin/kotlinperformancetester/src/main/kotlin/edu/imepac/kotlinperformancetester/services/EncosteService.kt
package edu.imepac.kotlinperformancetester.services

import edu.imepac.kotlinperformancetester.models.Encoste
import edu.imepac.kotlinperformancetester.models.Vagao
import edu.imepac.kotlinperformancetester.repositories.EncosteRepository
import jakarta.transaction.Transactional
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
class EncosteService {

    @Autowired
    private lateinit var encosteRepository: EncosteRepository

    fun findAll(): List<Encoste> {
        return encosteRepository.findAll()
    }

    @Transactional
    fun salvarTodos(encostes: List<Encoste>): List<Encoste> {
        return encosteRepository.saveAll(encostes)
    }
    fun create(encoste: Encoste): Encoste {
        return encosteRepository.save(encoste)
    }
}